get_defs.py

```diff
@@ -5,40 +5,45 @@ def api_call(word: str) -> list:
     
     trial = 0
     time.sleep(0.5)  #0.4 sleep with 2 attempts has had best results
+
     while trial < 2:
-        trial +=1
+
         try:
             url = f"https://api.dictionaryapi.dev/api/v2/entries/en/{word}"
             response = requests.get(url)
 
             if response.status_code == 200:
                     data = response.json()
-                    #print(f"Data found for {word}")
-                    return data
-            
+                    return data          
             else:
-                print(f"This was the error code: {response.status_code}")
-        except:
+                trial +=1
+                raise Exception(f"non-200 status code returned for {word}: {response.status_code}")
+            
+
+        except Exception:
             print(f"Bad Request - no relvant url found for {word}")
             return None
     
 
 #Returns a set of all found vocab types for a given word (e.g. {"verb", "noun"})
 def word_types(word):
     
-    word_types = set({})
+    word_types = set()
     current_call = api_call(word)
     try:
         if current_call != None:
             for items in current_call:
                 for mean_items in items['meanings']:
                     word_types.add(mean_items['partOfSpeech'])
-            if len(list(word_types)) > 0:
+            if len((word_types)) > 0:
                 return word_types
-            else:
-                return None
-    except:
+
+    except IndexError:
+        print(f"Index Error raised in word_types for {word}.")
         return None
```

Comment:
1. I really want to get you away from control-flow using None for exceptional behaviour as quickly as I can.

2. Note the error can be captured and printed using `except IndexError as indexError:` and then printed, or its message directly accessed for formatting in another error. The error can even be re-raised.

3. An IndexError is the kind of error which *practically never* has to occur - if a length check is done there should never be an IndexError raised. The only exception would be somehow not having access to the size or removing the check for extreme performance reasons.